### DISCUSSION TITLE
HomeKit/Unifi Doorbell Issue: Can't select HomeKit for G4 Doorbell Pro when set as "Doorbell"

### DISCUSSION THREAD

**Discussion by alexiscordova** [koush/scrypted#971] [Q&A] [answered]
I'm having a strange issue where my previously-working G4 Doorbell Pro no longer can register as a doorbell (so my HomePods can play the chime) in HomeKit; when set as "Camera", the camera functionality works but there are no options for doorbell chimes, naturally. While I have updated to the new iOS 16.1 (?) HomeKit architecture, that updated occurred a while ago and this issue only happened after I restarted my UDMP the other day.

### When G4 Doorbell Pro main camera is set as "Camera"
<img width="573" alt="Screenshot 2023-07-24 at 8 58 27 AM" src="https://github.com/koush/scrypted/assets/1877250/71d96e3a-bf39-4d4a-b107-a4d338ff400f">

When selecting the camera in the Unifi plugin, I see these options:
<img width="413" alt="Screenshot 2023-07-24 at 8 58 33 AM" src="https://github.com/koush/scrypted/assets/1877250/b1da46bf-22ab-4d28-bc41-e6b3b71cc81a">

I'm able to select HomeKit and see pairing information, settings, etc. "HomeKit" is selected, and the camera feed appears in the HomeKit app.
<img width="568" alt="Screenshot 2023-07-24 at 8 58 46 AM" src="https://github.com/koush/scrypted/assets/1877250/9c5f290b-ed26-4dfd-af2b-9195d48d9aff">

All doorbell features, specifically the "Chime on HomePod" feature, are gone.
![IMG_430701803F27-1](https://github.com/koush/scrypted/assets/1877250/8869bc37-dc7c-44a2-ac6a-d7e775d441d8)

### When G4 Doorbell Pro main camera is set as "Doorbell"
<img width="705" alt="Screenshot 2023-07-24 at 8 59 22 AM" src="https://github.com/koush/scrypted/assets/1877250/4dd6674b-8645-4b5f-83db-f23a2d1fb5d5">
When I switch the Doorbell Pro to "camera", I receive this notification.

After restarting the plugin, the Doorbell disappears from HomeKit
<img width="584" alt="Screenshot 2023-07-24 at 8 59 57 AM" src="https://github.com/koush/scrypted/assets/1877250/a17e51c1-8d04-4ecc-a544-64870ba66bd9">

<img width="572" alt="Screenshot 2023-07-24 at 9 00 02 AM" src="https://github.com/koush/scrypted/assets/1877250/b7c44de1-1144-403c-94b5-8def8d060e2d">

and the feed stops, naturally.

Any thoughts here? Am I missing something, or did something change?

**Comment by alexiscordova** [comment, marked as answer]
Turns out a regression was added sometime after Unifi Protect Plugin version 0.0.133. The following steps worked for me:
- Downgrade Unifi Protect Plugin to 0.0.133
- Restart Scrypted
- Remove the doorbell from HomeKit
- Re-pair the doorbell in HomeKit

I now have the functionality to set doorbell chimes again. cc @koush

**Comment by Bahweb** [comment]
@alexiscordova How can I downgrade I don't see that version available now in the plugin list..?

**Reply by alexiscordova** [reply]
The latest versions should fix this issue. I'm on the latest and the doorbell chimes through my HomePods

**Reply by Bahweb** [reply]
Yes I'm on latest 00.146 * 1.2.4 Homekit but they all stopped working recently.. tried all sorts to resolve it

**Comment by sup3rm3ga** [comment]
I am having this issue on 2 separate sites that both have the same setup with Scrypted and G4 Pro. Neither site actually functions correctly as far as I can tell. Ringing the doorbell no longer sends any Doorbell notification to HomeKit, and the camera does not function as a Doorbell. Have tried removing, re-adding, etc. Have not tried downgrading the Unifi protect plugin as stated, because that solution is so old.
